### PR TITLE
Remove "__default" key from CSS custom properties

### PR DESCRIPTION
**Pull request by fcisio**
This change follows my issue #1477 
Seems simple enough, let me know if you guys are down with this.

I had some type errors on my test, some color mode keys don't seem to allow for objects, only for color strings.

`primary?: CSS.Property.Color;`

### CODE REVIEW

File: packages/color-modes/src/custom-properties.ts [hunk 1]
```diff
@@ -1,8 +1,9 @@
 import { css, get, Theme, __internalGetUseRootStyles } from '@theme-ui/css'
 
-const toVarName = (key: string) => `--theme-ui-${key}`
+const stripKeyDefault = (key: string) => key.replace('-__default', '')
+const toVarName = (key: string) => `--theme-ui-${stripKeyDefault(key)}`
 const toVarValue = (key: string, value: string | number) =>
-  `var(${toVarName(key)}, ${value})`
+  `var(${toVarName(stripKeyDefault(key))}, ${value})`
```
Comment (hasparus) [inline packages/color-modes/src/custom-properties.ts]: I don't get this part.

Isn't `toVarValue` now

```
const toVarValue = (key: string, value: string | number) =>
  `var(`--theme-ui-${stripKeyDefault((stripKeyDefault(key)))`}, ${value})`
```

stripKeyDefault is called twice.

Comment (fcisio) [reply packages/color-modes/src/custom-properties.ts]: You're right, I did this pretty quickly I didn't realize they were fed into each other.
Just fixed it in a new commit.

**Comment by hasparus** [comment]
Okay it looks good, but I have a slight worry we have a few implicit dependencies between modules which are not covered with tests.

- [ ] Should `@theme-ui/custom-properties` also be updated?
- [x] Does this work with `getColor` from `@theme-ui/color`? (It should be fine, but I'll add tests for it.) 

---

> I had some type errors on my test, some color mode keys don't seem to allow for objects, only for color strings.
>
> primary?: CSS.Property.Color;

This is because many components from Theme UI components use `color: 'primary'`, so before `__default` was supported, if you had primary as object, you had to change their styles. Components predate both types and `__default`, and I wanted to suggest with TypeScript that "primary" as simple color will cause less problems down the road.

I'm gonna update types. I'll send another PR and ask for your reiew, okay?

**Review by hasparus — APPROVED** [review]

**Comment by hasparus** [comment]
As a side note for myself, [CI](
https://github.com/system-ui/theme-ui/blob/develop/.github/workflows/ci.yml) didn't run on this PR.

**Pull request merged**